model/resource.py

```diff
@@ -871,16 +920,19 @@ def record_exception(cls, representation, exception, traceback):
         representation.fetch_exception = traceback
 
     @classmethod
-    def post(cls, _db, url, data, max_age=None, response_reviewer=None):
+    def post(cls, _db, url, data, max_age=None, response_reviewer=None,
+             **kwargs):
         """Finds or creates POST request as a Representation"""
 
+        original_do_get = kwargs.pop('do_get', cls.simple_http_post)
+
         def do_post(url, headers, **kwargs):
             kwargs.update({'data' : data})
-            return cls.simple_http_post(url, headers, **kwargs)
+            return original_do_get(url, headers, **kwargs)
 
         return cls.get(
             _db, url, do_get=do_post, max_age=max_age,
-            response_reviewer=response_reviewer
+            response_reviewer=response_reviewer, **kwargs
```

Comment:
I needed to make this change to make NoveListAPI.lookup testable. I need to be able to specify a custom `do_get` for the test, even though `post` _also_ creates a custom `do_get`. I do this by having `post` _wrap_ any custom `do_get` it receives in a new, custom function.